tests/training/test_imputation.py

```diff
@@ -30,6 +30,30 @@ def test_imputer():
     assert expected2.equals(pred_fn(input_df2))
 
 
+def test_imputer_with_fill_value():
+    input_df = pd.DataFrame({
+        'col1': [10, 13, 10],
+        'col2': [50, 100, None],
+        'col3': [None, None, None]
+    })
+
+    df = pd.DataFrame({
+        'col1': [10.0, 13.0, 10.0],
+        'col2': [50.0, 100.0, 75.0],
+        'col3': [10.0, None, None]
+    })
+
+    expected = pd.DataFrame({
+        'col1': [10.0, 13.0, 10.0],
+        'col2': [50.0, 100.0, 75.0],
+        'col3': [10.0, -999, -999]
+    })
+
+    pred_fn, data, log = imputer(input_df, ["col1", "col2", "col3"], "mean", placeholder_value=-999)
+
+    assert expected.equals(pred_fn(df))
```

Comment:
instead of using `assert` and `equals`; it's best to use directly the [assertions provided by pandas](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.testing.assert_frame_equal.html). i'd change here to

```python
actual = pred_fn(df)
pd.testing.assert_frame_equal(expected, actual)
```

it also provides a more meaningful description whenever the assertion fails; instead of plain "assertion error".

Reply:
Thanks! ;D

It seams that just two tests in the while library use `assert_frame_equal` instead of a direct `assert`, so i think these shouldn't differ from the norm